lib/workers/repository/config-migration/branch/migrated-data.ts

```diff
@@ -0,0 +1,89 @@
+import detectIndent from 'detect-indent';
+import { migrateAndValidate } from '../../../../config/migrate-validate';
+import type { RenovateConfig } from '../../../../config/types';
+import { logger } from '../../../../logger';
+import { readLocalFile } from '../../../../util/fs';
+import { detectRepoFileConfig } from '../../init/merge';
+
+export class MigratedData {
+  constructor(
+    private readonly _content: string,
+    private readonly _fileName: string,
+    private readonly _indent: string
+  ) {}
+
+  getConfigContent(): string {
+    return this._content;
+  }
+
+  getConfigFileName(): string {
+    return this._fileName;
+  }
+
+  getIndent(): string {
+    return this._indent;
+  }
+}
+
+export class MigratedDataFactory {
+  // singleton
+  private static _data: MigratedData;
```

Comment:
https://google.github.io/styleguide/tsguide.html#class-members js's # are not recommended :|, should i use it anyway?

Reply:
we aren't following the google TS style guide. but if you follow it `_data` is wrong too.

they complain about js private fields because they aren't supported by versions older than es2015, that not an issue for us, as we already require es2020.

@rarkins do we like to follow google style guide? 🤷‍♂️

Reply:
> we aren't following the google TS style guide. but if you follow it _data is wrong too.

thats right, i might have misunderstood the conventions we are following. 
for example i saw that the _ prefix is used for method names (`GenericVersioningApi`) and assumed its acceptable for variables too. 

I do try to follow the google style guide when i can, still learning though.
learning and taking notes about the conventions mostly by the much appreciated CR we are getting, thats for sure.

ps, just to be clear here, do we use _ prefixes in private methods declarations?